src/common.hpp

```diff
@@ -70,6 +72,49 @@ namespace irods::http
         const authenticated_client_info* client_info{};
     }; // struct client_identity_resolution_result
 
+    class irods_connection // NOLINT(cppcoreguidelines-special-member-functions)
```

Comment:
Come up with a better name.
- `connection_facade`
- `connection_adapter`

This class needs to be part of the iRODS server. That can happen in a later release though. _(Capture this in an issue)_

Reply:
Renamed to `connection_facade`.